src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt

```diff
@@ -63,11 +63,8 @@ class IncentiveReviewsService(
     val offenders = deferredOffenders.await()
 
     val bookingIds = offenders.map(OffenderSearchPrisoner::bookingId)
-    val prisonerNumbers = offenders.map(OffenderSearchPrisoner::prisonerNumber)
-
-    val deferredPositiveCaseNotesInLast3Months = async { getCaseNoteUsage("POS", "IEP_ENC", prisonerNumbers) }
-    val deferredNegativeCaseNotesInLast3Months = async { getCaseNoteUsage("NEG", "IEP_WARN", prisonerNumbers) }
 
+    val deferredBehaviourCaseNotesSinceLastReview = async { behaviourService.getBehaviours(prisonerIepLevelRepository.findAllByBookingIdInOrderByReviewTimeDesc(bookingIds = bookingIds).toList()) }
```

Comment:
I'm not sure how big is the window of opportunity here but potentially someone may have no entries and therefore not be in this list (even tho they may have a some case notes maybe?)

Reply:
Another thing to bear in mind is that this list could be very very big, especially if the wing has a few prisoners which have been in prison for years and years and have a lot of records.

In theory we're only interested in the last 3 months, but of course adding that to the query means you would remove prisoners without a recent review from the list, and therefore their case notes would be returned.

Reply:
Every prisoner MUST have an incentive review, in which case they will all be considered -Good point though!